### DISCUSSION TITLE
Handling DMA level data

### DISCUSSION THREAD

**Discussion by SaiTarunY4** [facebookexperimental/Robyn#286] [Q&A] [answered]
Hi team,
How do you suggest we build an MMM at the DMA level using Robyn? Do we build a separate model for each DMA? Is there any other approach you would suggest we use? 
Note: There is some hierarchical structure w.r.t DMA i.e a set of DMAs roll up to a brand and so have a specific advertising strategy. 
We tend to conduct market tests (might be useful for calibration) by channel at DMA level.

**Comment by gufengzhou** [comment, marked as answer]
Hi, usually your dataset (panel data) requires a mixed model (or hierarchical model, or multilevel model, there're many names) to treat DMA as random effect. Robyn can't deal with that yet. Of course you can always build a separate model for each DMA if you have enough observations each. Our recommendation is column to row ratio of 1 to 10

**Reply by CJ2407** [reply]
@gufengzhou - Thanks for your response. Will building a model using Robyn still useful knowing that ideally we should have built it at DMA level? Also, is this work in progress at your end to introduce these features of building models with different group wise variables such as DMA, Distribution channel (Ecomm vs Retail), and product categories (e.g for Retail - Clothes vs Accessories vs Shoes) in Robyn?

**Reply by gufengzhou** [reply]
For now it probably won't work properly if you insert panel data. In the future, yes we do have this on the list but it's not very straight forward because Robyn would require regularised mixed model and we have to yet figure out how to do it best. So it won't be a short term goal

**Comment by Johannes96** [comment]
I have a similar issue. And thought about building a separate model to measure the relationship between the total ad-invest of individual shops vs total revenue. 
total revenue ~ Shop1_total_ad_spend + Shop2_total_ad_spend + ...

Now, in that special case the goal is to detect cannibalization effects between shops: Investing heavily in shop1 does not necessarily increase the total revenue but could also lead to just moving renue from other shops to shop1. 

What do you think about that idea?
